frontend/src/metabase/components/Badge/Badge.styled.tsx

```diff
@@ -43,7 +43,9 @@ export const MaybeLink = styled(RawMaybeLink)`
   }
 `;
 
-export const BadgeIcon = styled(Icon)<{ hasMargin: boolean }>`
+export const BadgeIcon = styled(Icon, {
+  shouldForwardProp: propName => propName !== "hasMargin",
```

Comment:
nit: I learned the other day that we have a util for this: https://github.com/metabase/metabase/blob/813258b9531e3cdac654864f4507aa8a27b4fb49/frontend/src/metabase/common/utils/doNotForwardProps.ts#L1-L3

Reply:
yup and I'd get rid of that util 😄